Sources/MongoSwift/BSON/BSONEncoder.swift

```diff
@@ -560,7 +561,13 @@ private class MutableDictionary: BSONValue {
         }
         set(newValue) {
             keys.append(key)
-            values.append(newValue)
+            if let newValue = newValue {
+                values.append(newValue)
```

Comment:
If the `else` branch below is simply going to search for the key and remove it, why should we bother with `keys.append(key)` for both cases instead of just doing it here?

Reply:
No reason to just put it in the non-deletion case, so I've done that.